types/atom/index.d.ts

```diff
@@ -5237,20 +5248,22 @@ export class TextBuffer {
      *  `::revertToCheckpoint` and `::groupChangesSinceCheckpoint`.
      *  @return A checkpoint ID value.
      */
-    createCheckpoint(): number;
+    createCheckpoint(options?: { selectionsMarkerLayer?: MarkerLayer }): number;
 
     /**
      *  Revert the buffer to the state it was in when the given checkpoint was created.
      *  @return A boolean indicating whether the operation succeeded.
      */
-    revertToCheckpoint(checkpoint: number): boolean;
+    revertToCheckpoint(checkpoint: number, options?: { selectionsMarkerLayer: MarkerLayer }):
+        boolean;
 
     /**
      *  Group all changes since the given checkpoint into a single transaction for
      *  purposes of undo/redo.
      *  @return A boolean indicating whether the operation succeeded.
      */
-    groupChangesSinceCheckpoint(checkpoint: number): boolean;
+    groupChangesSinceCheckpoint(checkpoint: number, options?:
+        { selectionsMarkerLayer: MarkerLayer }): boolean;
```

Comment:
Probably a good idea to move `selectionsMarkerLayer` object into a dedicated interface to avoid pain in the future?

Reply:
It would also enable us to easily document it:
> Restore snapshot of selections marker layer to given selectionsMarkerLayer.

Reply:
Within an interface, we couldn't really give much information about what the `selectionsMarkerLayer` property is actually doing. We'd need to give that information alongside each function using it.

For instance, `TextBuffer.createCheckpoint()` has this as the documentation for that property:
>When provided, skip taking snapshot for other selections markerLayers except given one.